spot_wrapper/spot_dance.py

```diff
@@ -41,22 +48,40 @@ def upload_animation(
                 tmp.write(animation_file_content)
             try:
                 animation_pb = convert_animation_file_to_proto(filename).proto
+                return self.upload_animation_proto(animation_pb)
             except Exception as e:
                 return (
                     False,
                     "Failed to convert animation file to protobuf message: {}".format(
                         e
                     ),
                 )
-            try:
-                self._logger.info("Uploading the name {}".format(animation_name))
-                upload_response = self._choreography_client.upload_animated_move(
-                    animation_pb, animation_name
+
+    def upload_animation_proto(self, animation: Animation) -> Tuple[bool, str]:
+        result = False
+        result_message = ""
+        try:
+            self._logger.info("Uploading the name {}".format(animation.name))
+            upload_response = self._choreography_client.upload_animated_move(
+                animation, animation.name
+            )
+            result = upload_response.status == UploadAnimatedMoveResponse.STATUS_OK
+            if result:
+                result_message = "Successfully uploaded"
+                if upload_response.warnings:
+                    result_message += " with warnings from validator {}".format(
+                        upload_response.warnings
+                    )
+            else:
+                result_message = (
+                    "Failed to upload animation with status {} and warnings: {}".format(
+                        upload_response.status, upload_response.warnings
+                    )
                 )
-            except Exception as e:
-                error_msg = "Failed to upload animation: {}".format(e)
-                return False, error_msg
-        return True, "Success"
+        except Exception as e:
+            result_message = "Failed to upload animation: {}".format(e)
+            return result, result_message
+        return result, result_message
```

Comment:
Other strings are using the f-string method rather than `.format`. For consistency probably good to switch these to fstrings as well.

Reply:
Done.